### PR TITLE
feat(unlock-app): Decent for swaps

### PR DESCRIPTION
**Pull request by julien51**
# Description

Until now we used Decent for the "crosschain purchase" scenario where the transaction would be sent from different network, but only paid on this network's native currency.
Since Uniswap broke their libraries, we are now also using Decent for "swap" scenario where we now have the following supported flows:

- Same chain, use ERC20 for a native token lock
- Same chain, use ERC20 for lock with a different ERC20
- Same chain, use native token for a lock using an ERC20
- Different chain, use native token for a native lock
- Different chain, use ERC20 for a native token lock
- Different chain, use ERC20 for lock with a different ERC20
- Different chain, use native token for a lock using an ERC20

# Issues

Fixes #
Refs #

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/checkout/main/Payment.tsx [hunk 1]
```diff
@@ -165,27 +164,15 @@ export function Payment({ checkoutService }: Props) {
 
   const canAfford = balance?.isGasPayable && balance?.isPayable
 
-  const { data: uniswapRoutes, isInitialLoading: isUniswapRoutesLoading } =
-    useUniswapRoutes({
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/main/Payment.tsx:169]: I will clean up on a different PR.

File: unlock-app/src/hooks/useCrossChainRoutes.ts [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+import { networks } from '@unlock-protocol/networks'
+import { BoxEvmChains } from '@decent.xyz/box-common'
+import { Token } from '@unlock-protocol/types'
+
+interface CrossChainRouteWithBalance extends CrossChainRoute {
+  resolvedAt: number
+  userTokenBalance: string
+}
+
+// TheBox returns BigInts as strings with a trailing 'n'
```
Comment (julien51) [inline unlock-app/src/hooks/useCrossChainRoutes.ts:18]: 🤦

File: unlock-app/src/components/interface/checkout/main/Payment.tsx [hunk 3]
```diff
@@ -425,22 +365,28 @@ export function Payment({ checkoutService }: Props) {
                   >
                     <div className="flex justify-between w-full">
                       <h3 className="font-bold">
-                        Pay with {route.symbol} on {route.networkName}
+                        Pay with {route.tokenPayment.symbol} on{' '}
```
Comment (clemsos) [inline unlock-app/src/components/interface/checkout/main/Payment.tsx:368]: that throws. an error when the lock is priced in native token. Added a fix

File: unlock-app/src/hooks/useCrossChainRoutes.ts (hunk 2 above)
Comment (clemsos) [reply unlock-app/src/hooks/useCrossChainRoutes.ts:18]: looking into it, yes its a real issue when parsing serialized values...

File: unlock-app/src/components/interface/checkout/main/Payment.tsx (hunk 3 above)
Comment (julien51) [reply unlock-app/src/components/interface/checkout/main/Payment.tsx]: ha! Good catch I will look at this gain.

**Pull request merged**